praw/models/reddit/removal_reasons.py

```diff
@@ -73,54 +66,93 @@ def delete(self):
 
         .. code-block:: python
 
-           reddit.subreddit('NAME').removal_reasons['141vv5c16py7d'].mod.delete()
+           reddit.subreddit('NAME').mod.removal_reasons['141vv5c16py7d'].delete()
 
         """
         url = API_PATH["removal_reason"].format(
             subreddit=self.subreddit, id=self.id
         )
-        self.subreddit._reddit.request("DELETE", url)
+        self._reddit.request("DELETE", url)
 
-    def update(self, message: str, title: str):
+    def update(
+        self, message: Optional[str] = None, title: Optional[str] = None
+    ):
         """Update the removal reason from this subreddit.
 
-        :param message: The removal reason's new message (required).
-        :param title: The removal reason's new title (required).
+        .. note:: Existing values will be used for any unspecified arguments.
+
+        :param message: The removal reason's new message.
+        :param title: The removal reason's new title.
 
         To update ``"141vv5c16py7d"`` from the subreddit ``"NAME"`` try:
 
         .. code-block:: python
 
-           reddit.subreddit('NAME').removal_reasons['141vv5c16py7d'].mod.update(
+           reddit.subreddit('NAME').mod.removal_reasons['141vv5c16py7d'].update(
                message='New message',
                title='New title')
 
         """
         url = API_PATH["removal_reason"].format(
             subreddit=self.subreddit, id=self.id
         )
-        data = {"message": message, "title": title}
-        self.subreddit._reddit.put(url, data=data)
+        data = {
+            name: getattr(self, name) if value is None else value
+            for name, value in {"message": message, "title": title}.items()
+        }
+        self._reddit.put(url, data=data)
 
 
 class SubredditRemovalReasons:
     """Provide a set of functions to a Subreddit's removal reasons."""
 
-    def __getitem__(self, reason_id: str) -> RemovalReason:
-        """Lazily return the Removal Reason for the subreddit with id ``reason_id``.
+    def __getitem__(
+        self, id: Union[str, int, slice]  # pylint: disable=redefined-builtin
```

Comment:
I think it's best not to rename this parameter, because technically it's changing the public interface of this method, even though ideally this would be invoked via `[id]` and so the parameter name wouldn't be used at all. If we were going to rename this, `id` is not a great name because (a) it's a builtin function, and (b) you use it as an index below, so the name is misleading.

Reply:
Submission and Comment objects use `id`, and `id` is what the Reddit API returns. I do think we can use reason_id as an alternate parameter, and deprecate it.